tests/mod.rs

```diff
@@ -267,3 +267,29 @@ fn enum_compact_meta_attribute_works() {
 		}
 	}
 }
+
+#[test]
+fn associated_type_bounds() {
+	trait Trait {
+		type EncodableType;
+		type NonEncodableType;
+	}
+
+	#[derive(Encode, Decode, Debug, PartialEq)]
+	struct Struct<T: Trait, Type> {
+		field: (Vec<T::EncodableType>, Type),
+	}
+
+	#[derive(Debug, PartialEq)]
+	struct TraitImplementor;
+
+	impl Trait for TraitImplementor {
+		type EncodableType = u32;
+		type NonEncodableType = std::marker::PhantomData<Self>;
```

Comment:
We provide an implementation of `Encode` for `PhantomData`. So this is an encodable type. Just add a random struct that represents the `NonEncodableType`.